x-pack/plugins/maps/public/components/layer_addpanel/view.js

```diff
@@ -4,167 +4,148 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { Component, Fragment } from 'react';
-import { ALL_SOURCES } from '../../shared/layers/sources/all_sources';
+import React, { Component } from 'react';
+import { SourceSelect } from './source_select/source_select';
+import { FlyoutFooter } from './flyout_footer';
+import { SourceEditor } from './source_editor';
+import { ImportEditor } from './import_editor';
 import {
-  EuiButton,
-  EuiButtonEmpty,
   EuiFlexGroup,
-  EuiFlexItem,
   EuiTitle,
-  EuiPanel,
-  EuiSpacer,
-  EuiCard,
-  EuiIcon,
   EuiFlyoutHeader,
-  EuiFlyoutFooter,
 } from '@elastic/eui';
-import { FormattedMessage } from '@kbn/i18n/react';
-import _ from 'lodash';
+import { i18n } from '@kbn/i18n';
 
 export class AddLayerPanel extends Component {
 
   state = {
     sourceType: null,
-    layer: null
+    layer: null,
+    panelDescription: i18n.translate('xpack.maps.addLayerPanel.selectSource',
+      { defaultMessage: 'Select source' }),
+    importView: false,
```

Comment:
`importView` does not need to stored on state and can just be derived with a function call since its just determined from Source type. That would simplify `_onSourceSelectionChange` since it could just accept the source type

Reply:
I think `layer_addpanel/view` could be diced up again into separate import & add components a step further than has already been done. Until we do that, `importView` is used throughout the component to render different parts for each workflow and is more descriptive to what the component is doing than calling `Source.isIndexingSource()` each time. If we want to dice up the workflows and and end up only making that call once to build a component, then I agree. Otherwise I'd rather leave it for now